boards/DIM/Src/Io/Io_SevenSegDisplays.c

```diff
@@ -0,0 +1,264 @@
+#include "App_SharedAssert.h"
+#include "App_SharedConstants.h"
+
+#include "Io_SevenSegDisplays.h"
+
+static_assert(HEX_DIGIT_0 == 0, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_1 == 1, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_2 == 2, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_3 == 3, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_4 == 4, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_5 == 5, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_6 == 6, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_7 == 7, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_8 == 8, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_9 == 9, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_A == 10, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_B == 11, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_C == 12, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_D == 13, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_E == 14, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_F == 15, "Hex enum must match its numeric value.");
+
+float Io_SevenSegDisplay_GetStateOfCharge(void)
+{
+    // TODO: Read the state of charge from CAN
+    return 0.0f;
+}
+
+void Io_SevenSegDisplay_SetLeftHexDigit(uint8_t hex_digit)
```

Comment:
With how we have the shift registers/ seven segs wired on the DIM we can not just change one seven segment at a time. The shift registers are daisy chained together you need to shift 3 x 8 bits over spi to update them all. We will need a LUT which will map each hex number to the corresponding bits needed to represent that number on seven segment display. After we have the bit representation of each digit, we would assign the elements in SPI tx data buffer, for example:

```
uint8_t spi_tx_data[3];
...
spi_tx_data[0] = right_hex_digit;
spi_tx_data[1] = middle_hex_digit;
spi_tx_data[2] = left_hex_digit;

HAL_SPI_Transmit(&hspi1, spi_tx_data, 3U, 100U);

Reply:
On second thought this function could be used to update the left hex digit in the tx buffer:
`spi_tx_data[2] = left_hex_digit;`
and then the state machine would periodically be calling 
`HAL_SPI_Transmit(&hspi1, spi_tx_data, 3U, 100U);`
which in turn would update that specific display?

Reply:
Yep that could be one way to do it. I think the approach is to write clean application-level code that is independent of hardware technicalities such as shift registers). The we will write IO-level code to accomdate the application-level code.

```
Task100Hz {
  App_SevenSegDisplays_Tick(); //Update spi_tx_data[0-2] using Io_SevenSegDisplay_Set{Left|Middle|Right}HexDigit
  Io_SevenSegDisplays_Tick(); // HAL_SPI_Transmit
}
```

Reply:
Ya, that sounds good then👍

Reply:
I'll tweak the code a bit to work with the SPI data better

Reply:
@markomarkusi Feel free to take a look at the code I rewrote and see if it makes more sense now with the shift register design